src/main/java/org/opensearch/knn/index/codec/util/KNNCodecUtil.java

```diff
@@ -111,8 +114,45 @@ public static String buildEngineFilePrefix(String segmentName) {
         return String.format("%s_", segmentName);
     }
 
+    /**
+     * Build a suffix of file name with provided field name and custom extension.
+     * In case where the given field name contains invalid characters for the file name, it will escape them to '.'.
+     * Note that, field name MUST NOT contain '.' in it.
+     * <p>
+     * Ex:
+     * With 'my_vector' as field name and '.hnswc' as an extension -> _0_2011_my_vector.hnswc
+     * With 'my vector' (blank) as field name and '.hnswc' as an extension -> _0_2011_my.vector.hnswc
+     * With 'my/vector' (it has '/') as field name and '.hnswc' as an extension -> _0_2011_my.vector.hnswc
```

Comment:
what is the behavior of Opensearch with such fields? does Opensearch adds `.` or they do something else?

Reply:
OpenSearch does not create its own file except lucene's segment file and lucene does not use field name in its file name but use random value.

Reply:
I think we just need to block the space in the field name. `.` is used for nested field.
For example
```
"mappings": {
    "properties": {
      "my_field": {
        "type": "nested",
        "properties": {
          "my_vector": {
            "type": "knn_vector",
            "dimension": 3
          }
      }
    }
  }
}
```
Then, file name will be `	_0_165_ my_field.my_vector.faissc`

Also, I was able to create field name using `.`. 
```
"mappings": {
    "properties": {
      "my_field.my_vector": {
        "type": "knn_vector",
        "dimension": 2,
        "method": {
          "name": "hnsw",
          "space_type": "innerproduct",
          "engine": "faiss"
        }
      }
    }
  }
```

Reply:
`a.b.c` will be splitted into three "actual" fields within Lucene.
So it will be translated to three fields where `a`, `b` and `c`.
Eventually, the field name to be passed down to `escapeSpaceInFieldName` does not contain dot ('.') character.

Reply:
For example, `my_field.my_vector` will call `escapeSpaceInFieldName` twice with each `my_field` and `my_vector`.

Reply:
KNN field might behave differently with lucene. Try to create knn index and test it locally. 
`escapeSpaceInFieldName` will be called only one time with field name `my_field.my_vector`.

Reply:
Can you share the query input you've tested?
I've tested it, and I got two fields to be passed into it.

Reply:
```
PUT /test_index
{
  "settings": {
    "index": {
      "knn": true,
      "number_of_shards": 1
      
    }
  },
  "mappings": {
    "properties": {
      "my_field.my_vector": {
        "type": "knn_vector",
        "dimension": 2,
        "method": {
          "name": "hnsw",
          "space_type": "innerproduct",
          "engine": "lucene"
        }
      }
    }
  }
}
```

```
PUT /_bulk?refresh=true
{ "index": { "_index": "test_index", "_id": "1" } }
{"my_field.my_vector":[-2.0, 2.0], "text": "a"}
{ "index": { "_index": "test_index", "_id": "2" } }
{"my_field.my_vector":[2.0, -2.0], "text": "d"}
```

```
POST /test_index/_search
{
  "query": {
    "knn": {
      "my_field.my_vector": {
        "vector": [
          -2.0,
          2.0
        ],
        "k": 2
      }
    }
  }
}
```

Reply:
I mean in `KNNWeight` not in `KNN80CompountFormat`